lib/document/query/validateQueryFactory.js

```diff
@@ -65,6 +70,15 @@ function validateQueryFactory(
           .map((field) => new NotIndexedFieldError(field)),
       );
 
+      // Check date fields
+      query.where
```

Comment:
we can validate it with json schema, just add one more condition to `oneOf` with `$createdAt|$updatedAt` field pattern